packages/dara-core/dara/core/interactivity/derived_variable.py

```diff
@@ -207,7 +208,9 @@ def _restore_pydantic_models(func: Callable[..., Any], *args):
                 if typ is not None and isclass(typ) and issubclass(typ, BaseModel) and arg is not None:
                     parsed_args.append(typ(**arg))
                     continue
```

Comment:
I think user-defined encoders should take precedence over our catch-all BaseModel handler. In some cases a BaseModel subclass might have a custom initializer where the default approach of `typ(**arg)` might not work.

Same in the other place restoring arg annotations